mesonbuild/mconf.py

```diff
@@ -38,11 +37,28 @@ class ConfException(mesonlib.MesonException):
 
 class Conf:
     def __init__(self, build_dir):
-        self.build_dir = build_dir
-        if not os.path.isdir(os.path.join(build_dir, 'meson-private')):
-            raise ConfException('Directory %s does not seem to be a Meson build directory.' % build_dir)
-        self.build = build.load(self.build_dir)
-        self.coredata = coredata.load(self.build_dir)
+        self.build_dir = os.path.abspath(os.path.realpath(build_dir))
+        if self.build_dir.endswith('/meson.build') or self.build_dir.endswith('\\meson.build') or self.build_dir == 'meson.build':
+            self.build_dir = os.path.dirname(self.build_dir)
+        self.build: build.Build = None
```

Comment:
This syntax won't work until 3.6, right? Also you're assigning None to a variable that you've declared as a `build.Build`.

Reply:
Sure, it makes more sense without the annotation.